src/common/lib/client/realtimechannel.ts

```diff
@@ -596,11 +586,21 @@ class RealtimeChannel extends Channel {
   }
 
   onMessage(message: ProtocolMessage): void {
+    // Note setting channel serial here rather than the switch below to be
+    // explicit (as with fall throughs its easy to update from the wrong
+    // message).
+    if (
+      message.action === actions.ATTACHED ||
+      message.action === actions.MESSAGE ||
+      message.action === actions.PRESENCE
+    ) {
+      this.setChannelSerial(message.channelSerial);
```

Comment:
Check if `channelSerial` is not empty, then only set it

Reply:
Thats handled by `setChannelSerial`

Reply:
As per RTL15b, use only if its set

Reply:
Yep handled by `setChannelSerial`:
```javascript
    if (channelSerial) {
      this.properties.channelSerial = channelSerial;
    }
```